sympy/core/function.py

```diff
@@ -707,6 +707,7 @@ class UndefinedFunction(FunctionClass):
     The (meta)class of undefined functions.
     """
     def __new__(mcl, name, **kwargs):
+        name=name.encode('ascii') # if user provides an UNICODE name
```

Comment:
`name.encode('ascii', 'ignore')` ignores error generated if non-ascii characters are found.
I thought it might be better, if it raises an error if it encounters such characters than ignore it.